### ISSUE TITLE
drs_stat is slow when applied to 100's of URLs

### ISSUE THREAD

**Issue by mtmorgan** [Bioconductor/AnVIL#38]
Intrinsic latency of http request. mclapply to parallelize (but not cross platform); snow / stability has its own latency issues.

**Comment by mtmorgan** [comment]
Another thing to explore is re-use of the connection 'handle' across calls, which should eliminate a lot of the latency.

**Comment by mtmorgan** [comment]
It looks like the handle is already re-used (e.g., from `?POST` and `?handle_pool`) and that parallel evaluation doesn't really help that much...
```
drs_eg_anvil <- c(
    "drs://dg.ANV0/975bd45f-f022-4fad-b9a2-3a00c3b8792c",
    "drs://dg.ANV0/00008531-03d7-418c-b3d3-b7b22b5381a0"
)

drs20 <- rep(drs_eg_anvil, 10)

options(mc.cores = 20L)
system.time(res <- drs_stat(drs20)) # elapsed time: 9s

options(mc.cores = 1L)
system.time(res <- drs_stat(drs20)) # elapsed time: 13s
```
@vjcitn do you seem something markedly different? Do you have a reproducible example?

**Comment by vjcitn** [comment]
I did see advantages.  I will open up a workspace to pursue this further.

**Comment by vjcitn** [comment]
```
> options(mc.cores=1)
> mark(drs_stat(drs_50[1:4]), memory=FALSE)
# A tibble: 1 x 13
  expression                min  median `itr/sec` mem_alloc `gc/sec` n_itr  n_gc total_time result      memory time     gc       
  <bch:expr>            <bch:t> <bch:t>     <dbl> <bch:byt>    <dbl> <int> <dbl>   <bch:tm> <list>      <list> <list>   <list>   
1 drs_stat(drs_50[1:4])     34s     34s    0.0294        NA        0     1     0        34s <tibble [4… <NULL> <bench_… <tibble …
> options(mc.cores=4)
> mark(drs_stat(drs_50[1:4]), memory=FALSE)
# A tibble: 1 x 13
  expression                min  median `itr/sec` mem_alloc `gc/sec` n_itr  n_gc total_time result      memory time     gc       
  <bch:expr>            <bch:t> <bch:t>     <dbl> <bch:byt>    <dbl> <int> <dbl>   <bch:tm> <list>      <list> <list>   <list>   
1 drs_stat(drs_50[1:4])   7.87s   7.87s     0.127        NA        0     1     0      7.87s <tibble [4… <NULL> <bench_… <tibble …
```
That factor is consistent with what I saw Monday.  But the general resolution seems much slower.  This just substitutes mclapply for lapply in drs_stat.

**Comment by mtmorgan** [comment]
closed with 77683f157f1252ad29259b78661f064667b0ac90
